lib/Common/Memory/MarkContext.inl

```diff
@@ -22,8 +22,12 @@ bool MarkContext::AddMarkedObject(void * objectAddress, size_t objectSize)
     RECYCLER_STATS_INTERLOCKED_INC(recycler, scanCount);
 
     MarkCandidate markCandidate;
-    markCandidate.obj = (void **) objectAddress;
+#if defined(_WIN64) && defined(_M_X64)
+    *(__m128i *)&markCandidate = _mm_set_epi64x(objectSize, (__int64)objectAddress);
+#else
+    markCandidate.obj = (void**)objectAddress;
     markCandidate.byteCount = objectSize;
```

Comment:
Yes. We have a load for the entire object(128bits) right after the 64bit stores of address and size.  This uneven size stores and load prevents the Stores from being forwarded to the Load
bypassing the caches. 
With the intrinsic the compiler will generate similar sized store and load (128bit) and helps to avoid the above issue and saves a few cycles.